### ISSUE TITLE
Text is not centered properly

### ISSUE THREAD

**Issue by mrianura** [mikke89/RmlUi#383]
Please take a look at that simple example:
![image](https://user-images.githubusercontent.com/15817792/204775508-73196249-3833-4f80-bddd-cf6daec99069.png)

As You can see, left margin is 31px meanwhile right is 35px.

RML:
```
<div>
 nooolooon
</div>
```
RCSS:
```
div
{
    margin-left: auto;
    margin-right: auto;
    width: 10%;
    height: 10%;
    background-color: #252526;
    text-align: center;
    text-transform: uppercase;
    color: white;
    text-family: LatoLatin-Regular;
    padding: 0;
}
```

**Comment by mrianura** [comment]
btw, no custom font manager used. Using default one - freeType

**Comment by mikke89** [comment]
Thanks for reporting.

I took a closer look at this one. It has to do with how trailing white space in elements are treated during layouting.

I made a test case and made a quick effort at solving this. While this fixes your case specifically, there is also at least one regression (see the screenshot below). The CSS rules for this behavior is actually quite complicated, [see the specs here for details](https://www.w3.org/TR/css-text-3/#white-space-processing). Not sure if we really want to do all of that work. If we do, then at least this will need more development and testing before it can be merged, to avoid any regressions.

For now, the simple workaround is to just not have any trailing white space inside your element. That is, `<div>nooolooon</div>` should be properly centered.

![image](https://user-images.githubusercontent.com/5490330/204941112-d1175ef0-3ed6-420f-a227-aa6d026eaa16.png)
